### ISSUE TITLE
DoS in http server code

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7564]
|<img alt="dalke's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| dalke reported|
|-|-|
|Trac ID|trac#287|
|Type|defect|
|Created|2003-09-26 14:28:04Z|

<details><summary>Searchable metadata</summary>

```
trac-id__287 287
type__defect defect
reporter__dalke dalke
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__ 
keywords__ 
time__1064586484000000 1064586484000000
changetime__1467629731230803 1467629731230803
version__ 
owner__ 
cc__itamarst cc__dalke
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="dalke's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|dalke commented|
|-|-|

```
#!html
<pre>
There's a DoS attack possible against the Twisted HTTP server code.
In short, there's no limit to the number of headers it will accept.  An 
attacker can fill up memory, or just keep it close to running out of 
memory and cause problems with other processes.

Here's demo attack code

http://twistedmatrix.com/pipermail/twisted-python/2003-July/
004825.html

</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst commented|
|-|-|

```
#!html
<pre>
Related bugs are the file upload issues.

</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst commented|
|-|-|

```
#!html
<pre>
Fixed in CVS.

</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_5"></a>|GitHub commented|
|-|-|

In changeset 40c4268b4c8a2dffc8f8cc2a9ed390d27ba7bcd2
```
#!CommitTicketReference repository="" revision="40c4268b4c8a2dffc8f8cc2a9ed390d27ba7bcd2"
Merge pull request #287 from twisted/7807-rodrigc-extrasrequire-py3

Author: rodrigc
Reviewer: Lukasa
Fixes: #7807

Fix on Python 3:  pip install -e ".[dev]"
```
